src/features/Authentication/authenticationApi.ts

```diff
@@ -1,92 +1,203 @@
+import * as D from 'io-ts/Decoder';
 import { createApi } from '@reduxjs/toolkit/query/react';
+import { pipe } from 'fp-ts/lib/function';
+import { t } from 'i18next';
+import toast from 'react-hot-toast';
+
 import { parseAndTransformTo, refreshingBaseQuery } from '@/utils/http';
-import * as D from 'io-ts/Decoder';
 
 export const role = D.literal('mentee', 'mentor', 'admin');
 
 const userCodec = D.struct({
   account_id: D.string,
   active: D.boolean,
-  created: D.string,
   display_name: D.string,
   id: D.string,
   role: role,
-  updated: D.string,
 });
 
 const accountCodec = D.struct({
   active: D.boolean,
-  created: D.string,
   email: D.string,
   id: D.string,
   login_name: D.string,
   role: role,
-  updated: D.string,
 });
 
-const myuserResponse = D.struct({
+const mentorCodec = D.struct({
+  account_id: D.string,
+  active: D.boolean,
+  birth_year: D.number,
+  communication_channels: D.array(D.string),
+  display_name: D.string,
+  gender: D.string,
+  id: D.string,
+  is_vacationing: D.boolean,
+  languages: D.array(D.string),
+  region: D.string,
+  skills: D.array(D.string),
+  status_message: D.string,
+  story: D.string,
+  user_id: D.string,
+});
+
+const commonResponse = D.struct({
   account: accountCodec,
   user: userCodec,
 });
 
-type UserResponse = D.TypeOf<typeof myuserResponse>;
+const mentorResponse = D.partial({
+  mentor: mentorCodec,
+});
+
+const myuserResponse = pipe(commonResponse, D.intersect(mentorResponse));
+
 export type UserRole = D.TypeOf<typeof role>;
 
-type AppUser = {
-  active: boolean;
-  displayName: string;
-  email: string;
-  loginName: string;
-  userId: string;
-  userRole: UserRole;
+export type Account = D.TypeOf<typeof accountCodec>;
+export type MentorUser = D.TypeOf<typeof mentorCodec>;
```

Comment:
Here too, we already have a MentorUser or Mentor somewhere. It might become a pain in the ass if there is multiple same ones.

Reply:
Comment above, but done now.